src/server/server.zig

```diff
@@ -23,12 +25,62 @@ const command = @import("command/_command.zig");
 pub const WorldEditData = struct {
 	selectionPosition1: ?Vec3i = null,
 	selectionPosition2: ?Vec3i = null,
-	clipboard: ?main.blueprint.Blueprint = null,
+	clipboard: ?Blueprint = null,
+	undoHistory: History = .{},
+
+	const History = struct {
+		const DLList = std.DoublyLinkedList(Value);
+		changes: DLList = .{},
+		maxCapacity: u32 = 16,
```

Comment:
Could be a constant. I don't see a reason this should be runtime-modifiable.

Reply:
Interesting. I was aiming for making it configurable, because when I use worldedit locally I can afford really big history, but a server can't do that for every player.

Reply:
(Also in terms of affording a big history, 1000 actions per player does not seem like an unreasonable default)

Reply:
Well, 1000 actions is roughly 1MB when using sphere brush with radius of 3 blocks, for radius 5 it is already above 4MB, in world edit max radius is by default 6 (or maybe its diameter, not sure rn). Realistically players are unlikely to saturate history with just `/paste` and `/set`, although those have potential to generate much larger entries.

Reply:
I calculated 131 MB for 32×32×32 paste operations. Even that is nothing, you'd need 100 players doing that to get anywhere close to the available RAM.